chitchat/src/state.rs

```diff
@@ -961,49 +1140,72 @@ mod tests {
-        node_state1.set_with_version("key_c", "value_c", 3);
-        stale_nodes.offer(&node1, &node_state1, 1u64);
+        let node2 = ChitchatId::for_local_test(10_002);
+        let node3 = ChitchatId::for_local_test(10_003);
+        let node4 = ChitchatId::for_local_test(10_004);
+        let node5 = ChitchatId::for_local_test(10_005);
+        let node6 = ChitchatId::for_local_test(10_006);
+
+        {
+            let mut node_state1 = cluster_state.node_state_mut(&node1);
+            node_state1.set_with_version("key_a", "value_a", 1);
+            node_state1.set_with_version("key_b", "value_b", 2);
+            node_state1.set_with_version("key_c", "value_c", 3);
+        }
+
         // 2 stale values.
+        {
+            let mut node_state2 = cluster_state.node_state_mut(&node2);
+            node_state2.set_with_version("key_a", "value", 1);
+            node_state2.set_with_version("key_b", "value_b", 2);
+            node_state2.set_with_version("key_c", "value_c", 5);
+        }
 
-        let node2 = ChitchatId::for_local_test(10_002);
-        let mut node_state2 = NodeState::for_test();
-        node_state2.set_with_version("key_a", "value", 1);
-        node_state2.set_with_version("key_b", "value_b", 2);
-        node_state2.set_with_version("key_c", "value_c", 5);
-        stale_nodes.offer(&node2, &node_state2, 2u64);
         // 1 stale value.
+        {
+            let mut node_state3 = cluster_state.node_state_mut(&node3);
+            node_state3.set_with_version("key_a", "value_a", 1);
+            node_state3.set_with_version("key_b", "value_b", 2);
+            node_state3.set_with_version("key_c", "value_c", 3);
+        }
 
-        let node3 = ChitchatId::for_local_test(10_003);
-        let mut node_state3 = NodeState::for_test();
-        node_state3.set_with_version("key_a", "value_a", 1);
-        node_state3.set_with_version("key_b", "value_b", 2);
-        node_state3.set_with_version("key_c", "value_c", 3);
-        stale_nodes.offer(&node3, &node_state3, 7u64);
         // 0 stale values.
+        {
+            let mut node_state4 = cluster_state.node_state_mut(&node4);
+            node_state4.set_with_version("key_a", "value_a", 1);
+            node_state4.set_with_version("key_b", "value_b", 2);
+            node_state4.set_with_version("key_c", "value_c", 5);
+            node_state4.set_with_version("key_d", "value_d", 7);
+        }
 
-        let node4 = ChitchatId::for_local_test(10_004);
-        let mut node_state4 = NodeState::for_test();
-        node_state4.set_with_version("key_a", "value_a", 1);
-        node_state4.set_with_version("key_b", "value_b", 2);
-        node_state4.set_with_version("key_c", "value_c", 5);
-        node_state4.set_with_version("key_d", "value_d", 7);
-        stale_nodes.offer(&node4, &node_state4, 1);
+        // 0 stale values
+        {
+            cluster_state.node_state_mut(&node5);
+        }
 
         // 3 stale values
```

Comment:
I don't understand where the stale value count come from. Before node5 and node6 had respectively 3 and 0 (instead of 0 and 3) stale values. I don't see an obvious reason this changed